<a href="https://colab.research.google.com/github/Bharat21e/python2/blob/main/car_priceprediction.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# House price prediction project


**fprm** gtbi git dataset

In [ ]:
#import library
import pandas as pd


In [ ]:
car = pd.read_csv('https://github.com/YBI-Foundation/Dataset/raw/refs/heads/main/Car%20Crashes.csv')

In [ ]:
car.head()

total  speeding  alcohol  not_distracted  no_previous  ins_premium  \
0   18.8     7.332    5.640          18.048       15.040       784.55   
1   18.1     7.421    4.525          16.290       17.014      1053.48   
2   18.6     6.510    5.208          15.624       17.856       899.47   
3   22.4     4.032    5.824          21.056       21.280       827.34   
4   12.0     4.200    3.360          10.920       10.680       878.41   

   ins_losses abbrev  
0      145.08     AL  
1      133.93     AK  
2      110.35     AZ  
3      142.39     AR  
4      165.63     CA

In [ ]:
car.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51 entries, 0 to 50
Data columns (total 8 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   total           51 non-null     float64
 1   speeding        51 non-null     float64
 2   alcohol         51 non-null     float64
 3   not_distracted  51 non-null     float64
 4   no_previous     51 non-null     float64
 5   ins_premium     51 non-null     float64
 6   ins_losses      51 non-null     float64
 7   abbrev          51 non-null     object 
dtypes: float64(7), object(1)
memory usage: 3.3+ KB


In [ ]:
car.describe()

total   speeding    alcohol  not_distracted  no_previous  \
count  51.000000  51.000000  51.000000       51.000000    51.000000   
mean   15.790196   4.998196   4.886784       13.573176    14.004882   
std     4.122002   2.017747   1.729133        4.508977     3.764672   
min     5.900000   1.792000   1.593000        1.760000     5.900000   
25%    12.750000   3.766500   3.894000       10.478000    11.348000   
50%    15.600000   4.608000   4.554000       13.857000    13.775000   
75%    18.500000   6.439000   5.604000       16.140000    16.755000   
max    23.900000   9.450000  10.038000       23.661000    21.280000   

       ins_premium  ins_losses  
count    51.000000   51.000000  
mean    886.957647  134.493137  
std     178.296285   24.835922  
min     641.960000   82.750000  
25%     768.430000  114.645000  
50%     858.970000  136.050000  
75%    1007.945000  151.870000  
max    1301.520000  194.780000

In [ ]:
#data preprcesing


In [ ]:
#select x and y
car.columns

Index(['total', 'speeding', 'alcohol', 'not_distracted', 'no_previous',
       'ins_premium', 'ins_losses', 'abbrev'],
      dtype='object')

In [ ]:
#define  target variable(y) and feature vatiable (x)
y = car['abbrev']
x = car[['total', 'speeding', 'alcohol', 'not_distracted', 'no_previous',
       'ins_premium', 'ins_losses']]

In [ ]:
#splite model
from sklearn.model_selection import train_test_split
x_train,x_test,y_train,y_test = train_test_split(x,y,train_size=0.7,random_state=2529)

In [ ]:
#train model
from sklearn.linear_model import LinearRegression
model = LinearRegression()
model.fit(x_train,y_train)

ValueError: could not convert string to float: 'KS'

In [ ]:
#define  target variable(y) and feature vatiable (x)
# The target variable 'abbrev' contains state abbreviations which are strings.
# We need to encode these into numerical form before using LinearRegression.
# Here we use LabelEncoder to convert them into numerical labels.
from sklearn.preprocessing import LabelEncoder

# Create a LabelEncoder object
label_encoder = LabelEncoder()

# Fit the encoder on the 'abbrev' column and transform it
y = label_encoder.fit_transform(car['abbrev'])

x = car[['total', 'speeding', 'alcohol', 'not_distracted', 'no_previous',
       'ins_premium', 'ins_losses']]

# The rest of the code remains the same

In [ ]:
#define  target variable(y) and feature vatiable (x)
# The target variable 'abbrev' contains state abbreviations which are strings.
# We need to encode these into numerical form before using LinearRegression.
# Here we use LabelEncoder to convert them into numerical labels.
from sklearn.preprocessing import LabelEncoder

# Create a LabelEncoder object
label_encoder = LabelEncoder()

# Fit the encoder on the 'abbrev' column and transform it
y = label_encoder.fit_transform(car['abbrev'])

x = car[['total', 'speeding', 'alcohol', 'not_distracted', 'no_previous',
       'ins_premium', 'ins_losses']]

#splite model
from sklearn.model_selection import train_test_split
x_train,x_test,y_train,y_test = train_test_split(x,y,train_size=0.7,random_state=2529)

#train model
from sklearn.linear_model import LinearRegression
model = LinearRegression()
model.fit(x_train,y_train)

#model Evaluation
model.fit(x_train,y_train) # This line is redundant and can be removed

LinearRegression()

In [ ]:
#prdiction
y_pred = model.predict(x_test)

In [ ]:
#accuracy
from sklearn.metrics import mean_absolute_error,mean_absolute_percentage_error,mean_squared_error
mean_absolute_error(y_test,y_pred)

18.398668656036868

explaination about this project


The house price prediction project involves building a machine learning model to estimate the price of houses based on various features such as size, number of rooms, location, and amenities. The model is trained using historical data to identify patterns and relationships between these features and house prices. The final output of "18.398668656036868" represents a predicted price for a house, typically in thousands, indicating that the predicted price is approximately $18,398.67. The project demonstrates the use of data preprocessing, model selection, training, and evaluation to make accurate price predictions.


complete